### PR TITLE
 Incorporate now for named companies after NR is approved

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#16942

*Description of changes:*
- Added incorporate now button for named companies after the NR has been approved.

I tested all the possible scenarios (BC, BEN, CC, ULC). I tried implementing the button as clean as possible with minimal code additions by building upon previous work.
UXPin: https://preview.uxpin.com/a86dccdd20ac62828f965c5eea10f81b828a9853#/pages/163136223/simulate/sitemap

### Example for CC:
After NR is Approved:
NR approved

Redirect:
incorporate with NR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://namerequest-dev--pr-627-58y7qy7d.web.app

File: src/components/common/bullets-colin-link.vue
Comment (JazzarKarim) [inline src/components/common/bullets-colin-link.vue:98]: Moving this function to Common Mixin. It makes sense since it'll come in handy for even further work later on.

File: src/components/existing-request/existing-request-display.vue
Comment (JazzarKarim) [inline src/components/existing-request/existing-request-display.vue:210]: This button is now in the `NRApprovedGrayBoy` above. I'm following the UXPin. It has similar behavior as the `Register Your Business` button

File: src/components/existing-request/existing-request-display.vue
Comment (JazzarKarim) [inline src/components/existing-request/existing-request-display.vue:418]: We'll be showing the button if the current entity type is supported (from FF), new business, and if the NR has been approved.

File: src/mixins/nr-affiliation-mixin.ts
Comment (JazzarKarim) [inline src/mixins/nr-affiliation-mixin.ts:138]: This'll behave the same as before but now instead of a hard-coded 'BEN', we're checking for FF values.

**Comment by pwei1018** [comment]
Temporary Url for review: https://namerequest-dev--pr-627-58y7qy7d.web.app

File: src/components/existing-request/nr-approved-gray-box.vue
Comment (severinbeauvais) [inline src/components/existing-request/nr-approved-gray-box.vue:24]: Whoa, way too easy! 😃

File: src/components/existing-request/existing-request-display.vue
Comment (severinbeauvais) [reply src/components/existing-request/existing-request-display.vue:418]: I think you can remove line 416 since line 417 will cover it.

Where did you copy the "if approved" logic from? It may be complicated that this (ie, conditional approval). Not certain but asking if you've looked into it.

File: src/components/existing-request/existing-request-display.vue
Comment (JazzarKarim) [reply src/components/existing-request/existing-request-display.vue:201]: I just checked, you're 100% right. Awesome catch Sev!

File: src/components/existing-request/existing-request-display.vue
Comment (JazzarKarim) [reply src/components/existing-request/existing-request-display.vue:418]: The logic here is almost similar to the `showRegisterButton` on line 422 with the difference being instead of `isFirm`, it's `isSupportedEntity`. I do agree with you on the first point. I was just using the similar condition to the existing one (`showRegisterButton`).

Comment (severinbeauvais) [reply src/components/existing-request/existing-request-display.vue:416]: Please call out for some extra testing on acceptable NR states to incorporate from (and/or confirm what's done in Auth Web for IA NRs).

Otherwise, my concerns are satisfied here.

**Comment by pwei1018** [comment]
Temporary Url for review: https://namerequest-dev--pr-627-58y7qy7d.web.app

File: src/mixins/nr-affiliation-mixin.ts
Comment (severinbeauvais) [reply src/mixins/nr-affiliation-mixin.ts]: Your new code looks cleaner, for sure 👍

**Review by severinbeauvais — COMMENTED** [review]
LGTM. As always, please get some other reviews before merging.

File: src/components/existing-request/existing-request-display.vue
Comment (JazzarKarim) [reply src/components/existing-request/existing-request-display.vue:416]: I don't think so. I took a look and this should be OK.

Keep in mind that the component takes a prop that disables the button when the NR was approved but still in processing (5 minutes or so).

**Review by leodube-aot — APPROVED** [review]
LGTM Karim!

**Review by chenhongjing — APPROVED** [review]
LGTM 🐱

**Comment by JazzarKarim** [comment]
@severinbeauvais Sev, can you approve please? I need an approval from someone with write access to this repo to merge 😁

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**
